### ISSUE TITLE
StackOverflowError in getTextSize() on line 306

### ISSUE THREAD

**Issue by tom91136** [grantland/android-autofittextview#11]
While trying with various Japanese strings, I've encountered a StackOverflowError:
```
java.lang.StackOverflowError
            at android.text.TextUtils.getChars(TextUtils.java:73)
            at android.text.MeasuredText.setPara(MeasuredText.java:106)
            at android.text.StaticLayout.generate(StaticLayout.java:239)
            at android.text.StaticLayout.<init>(StaticLayout.java:140)
            at android.text.StaticLayout.<init>(StaticLayout.java:90)
            at android.text.StaticLayout.<init>(StaticLayout.java:68)
            at android.text.StaticLayout.<init>(StaticLayout.java:48)
            at me.grantland.widget.AutofitTextView.getTextSize(AutofitTextView.java:297)
            at me.grantland.widget.AutofitTextView.getTextSize(AutofitTextView.java:306)
            at me.grantland.widget.AutofitTextView.getTextSize(AutofitTextView.java:306)
            at me.grantland.widget.AutofitTextView.getTextSize(AutofitTextView.java:306)
            at me.grantland.widget.AutofitTextView.getTextSize(AutofitTextView.java:306)
            at me.grantland.widget.AutofitTextView.getTextSize(AutofitTextView.java:306)
            // goes on repeating ......
```

getTextSize() gets caught in an infinite loop while doing a binary search looking for the best size.
The String I'm trying to display is this:
```
キスアト
```
Note that this is not the only String that crashes, any random Japanese Strings could cause a crash.
Another example
````
美少女写真館 番外編 ＯＵＴＳＩＤＥ ＳＴＯＲＹ
````
This seems to be related to encoding....

**Comment by grantland** [comment]
The repro looks to be setting/pasting text with more newline characters than `maxLines`, as long as `maxLines` is greater than 1.

**Comment by grantland** [comment]
Please review fix here: #32
